test/OrchardCore.Tests/Localization/PortableObjectStringLocalizerTests.cs

```diff
@@ -199,30 +202,32 @@ public void LocalizerReturnsOriginalTextForPluralIfTranslationDoesntExist(string
             SetupDictionary("cs", new[] {
                 new CultureDictionaryRecord("ball", "míč", "míče", "míčů"),
             });
-            var localizer = new PortableObjectStringLocalizer(null, _localizationManager.Object, true, _logger.Object);
-            CultureInfo.CurrentUICulture = new CultureInfo("cs");
-            var translation = localizer.Plural(count, "car", "cars");
 
-            Assert.Equal(expected, translation);
+            var localizer = new PortableObjectStringLocalizer(null, _localizationManager.Object, true, _logger.Object);
+            using (CultureScope.Create("cs"))
+            {
+                var translation = localizer.Plural(count, "car", "cars");
+                Assert.Equal(expected, translation);
+            }
         }
 
         [Theory]
         [InlineData("zh-Hans", "球", 1, new string[] { "球" })]
         [InlineData("zh-Hans", "球", 2, new string[] { "球" })]
         public void LocalizerReturnsCorrectTranslationForPluralIfNoPluralFormsSpecified(string culture, string expected, int count, string[] translations)
         {
-            var currentCulture = CultureInfo.GetCultureInfo(culture);
-            CultureInfo.CurrentUICulture = currentCulture;
+            using (var cultureScope = CultureScope.Create(culture))
+            {
+                // using DefaultPluralRuleProvider to test it returns correct rule
+                TryGetRuleFromDefaultPluralRuleProvider(cultureScope.Culture, out var rule);
```

Comment:
> But just one thing, i don't see any usage of these 2 public properties

It's used here

Reply:
First, based on the removed line, here maybe `cultureScope.UICulture`

Okay, so you also use the culture scope as a CultureInfo helper to get a cultureInfo from a string

Then because the culture scope set the async local cultures you can just use `CultureInfo.CurrentUICulture`, **that's the goal of this scope** to set cultures that can be retrieved from everywhere called in the context of this scope

            using (CultureScope.Create(CultureInfo.GetCultureInfo(culture)))
            {
                // using DefaultPluralRuleProvider to test it returns correct rule
                TryGetRuleFromDefaultPluralRuleProvider(CultureInfo.CurrentUICulture, out var rule);

Or if you want to keep the helper part creating a cultureInfo from a string

            using (CultureScope.Create(culture))
            {
                // using DefaultPluralRuleProvider to test it returns correct rule
                TryGetRuleFromDefaultPluralRuleProvider(CultureInfo.CurrentUICulture, out var rule);

Reply:
> First, based on the removed line, here maybe cultureScope.UICulture

Ok, but it's the same in this case

> Okay, so you also use the culture scope as a CultureInfo helper to get a cultureInfo from a string

There's already an overload to do that

Reply:
Okay will see tomorrow if it is not yet merged ;)

The main thing is that you are using a scope to set the static async local CultureInfo properties, and then for your tests in place of retrieving these values from this same async local CultureInfo you are retrieving them from the scope you just created, so here you just test the properties of this scope, you are not testing what it is intended to do.

E.g. when using this scope for liquid rendering (when not in a real view context), the template rendering will still retrieve the cutures from the static CultureInfo, that's the goal

Reply:
I may add some unit tests